### ISSUE TITLE
Autojoin a channel/room by using the contact list

### ISSUE THREAD

**Issue by georgehazan** [miranda-ng/miranda-ng#331]
**Reported by pirlouy on 22 Sep 2014 17:57 UTC**
Right now, at login, if you want to autojoin a channel/room, on IRC you have to configure advanced tab (perform /join #blabla), for XMPP you rely on server (bookmarks). If you use Gtalk for XMPP (for example) you can't autojoin channel.

I think it could be nice to use clist to autojoin a channel automatically, since you can do it by double clicking the contact/room.
I think a setting could be incorporated in:
right click the contact/room > "Contact list settings"... > checkbox "autojoin room at login"

**Comment by georgehazan** [comment]
**Comment by watcher on 24 Sep 2014 07:22 UTC**
When a server refuses to implement conference auto-join feature as well as bookmarks, than user that wants these features should change this server for one that does - and there is not shortage of Jabber servers out there.We already have 2 checkboxes for auto-connecting conferences in Jabber and it has been enormous burden on supporters to explain this combination time after time... Now the tired checkbox?!  Thing is - such "features" give people wrong ideas of how things work in protocols and they get totally confused on other clients, this is pure QIP way of doing things.

robyer, 24.09.2014 10:09:19:
also when you start miranda and connect account..

robyer, 10:09:24:
it must go through ALL contacts

robyer, 10:09:29:
check which one is for this account

robyer, 10:09:36:
and which one has set this flag

This is too much for me to take. Same goes to "convert to chat", people started keeping conferences in roster instead of bookmarks, then on new profile just convert them, no harm done. But switch to other XMPP client an you are stuck with bunch of unusable contacts that you can't turn to conferences.

Bottom line - we do have a way to enter the conference from double clicking on the conference contact in contact list. Auto-join should be supported by server. Let's not invent the wheel, turn to QIP and confuse new users about how things work.

All said above is my humble opinion, but also advice to coders - please think twice before you do it - is it all worth it and will it fix things o break more of them.

**Comment by georgehazan** [comment]
**Comment by wishmaster on 8 Oct 2014 06:48 UTC**
Replying to [watcher](comment:1):
> When a server refuses to implement conference auto-join feature as well as bookmarks, than user that wants these features should change this server for one that does - and there is not shortage of Jabber servers out there.We already have 2 checkboxes for auto-connecting conferences in Jabber and it has been enormous burden on supporters to explain this combination time after time...
Well, the main problem is that this isn't obvious to the user that this is server-dependant. So if someone uses a server that doesn't support it, it looks like a bug to him.
> Now the tired checkbox?!  Thing is - such "features" give people wrong ideas of how things work in protocols and they get totally confused on other clients, this is pure QIP way of doing things.
Well, we wouldn't need a third checkbox, we could reuse the existing checkbox for servers that don't support bookmarks.

> robyer, 24.09.2014 10:09:19:
> also when you start miranda and connect account..
> 
> robyer, 10:09:24:
> it must go through ALL contacts
> 
> robyer, 10:09:29:
> check which one is for this account
> 
> robyer, 10:09:36:
> and which one has set this flag
> 
> This is too much for me to take.
Why too much? We would only have to go through contacts of that account, not through all, and only for servers that don't support it.
> Same goes to "convert to chat", people started keeping conferences in roster instead of bookmarks, then on new profile just convert them, no harm done. But switch to other XMPP client an you are stuck with bunch of unusable contacts that you can't turn to conferences.
I agree, "convert to chat" should be removed.
